packages/flutter_tools/lib/src/asset.dart

```diff
@@ -622,39 +635,47 @@ class ManifestAssetBundle implements AssetBundle {
     return deferredComponentsAssetVariants;
   }
 
-  DevFSStringContent _createAssetManifest(
+  Map<String, List<String>> _createAssetManifest(
     Map<_Asset, List<_Asset>> assetVariants,
     Map<String, Map<_Asset, List<_Asset>>> deferredComponentsAssetVariants
   ) {
-    final Map<String, List<String>> jsonObject = <String, List<String>>{};
-    final Map<_Asset, List<String>> jsonEntries = <_Asset, List<String>>{};
+    final Map<String, List<String>> manifest = <String, List<String>>{};
+    final Map<_Asset, List<String>> entries = <_Asset, List<String>>{};
     assetVariants.forEach((_Asset main, List<_Asset> variants) {
-      jsonEntries[main] = <String>[
+      entries[main] = <String>[
         for (final _Asset variant in variants)
           variant.entryUri.path,
       ];
     });
     if (deferredComponentsAssetVariants != null) {
       for (final Map<_Asset, List<_Asset>> componentAssets in deferredComponentsAssetVariants.values) {
         componentAssets.forEach((_Asset main, List<_Asset> variants) {
-          jsonEntries[main] = <String>[
+          entries[main] = <String>[
             for (final _Asset variant in variants)
               variant.entryUri.path,
           ];
         });
       }
     }
-    final List<_Asset> sortedKeys = jsonEntries.keys.toList()
+    final List<_Asset> sortedKeys = entries.keys.toList()
         ..sort((_Asset left, _Asset right) => left.entryUri.path.compareTo(right.entryUri.path));
     for (final _Asset main in sortedKeys) {
       final String decodedEntryPath = Uri.decodeFull(main.entryUri.path);
-      final List<String> rawEntryVariantsPaths = jsonEntries[main]!;
+      final List<String> rawEntryVariantsPaths = entries[main]!;
       final List<String> decodedEntryVariantPaths = rawEntryVariantsPaths
         .map((String value) => Uri.decodeFull(value))
         .toList();
-      jsonObject[decodedEntryPath] = decodedEntryVariantPaths;
+      manifest[decodedEntryPath] = decodedEntryVariantPaths;
     }
-    return DevFSStringContent(json.encode(jsonObject));
+    return manifest;
+  }
+
+  DevFSByteContent _createAssetManifestBinary(
+    Map<String, List<String>> assetManifest
+  ) {
+    const StandardMessageCodec codec = StandardMessageCodec();
+    final ByteData result = codec.encodeMessage(assetManifest)!;
```

Comment:
This format is the same as the old format, whereas I believe the design document lists a few options for expanding this to support more fields (locale/light dark mode) by making the entries objects.

Reply:
Yeah. Yesterday, I realized I totally skipped that part. Added a note about that in the PR description. I'll work on it.